shark-ai/export_and_serve.py

```diff
@@ -0,0 +1,412 @@
+  -a, --artifact-dir ARTIFACT_DIR   the location where the artifacts (sharded weights) should be saved. Defaults to EXPORT_DIR/artifacts/
+  -s, --shard                       shard the weight file in tp8 mode and export to MLIR.
+  -b, --batch-sizes BATCH_SIZES     batch sizes for export. Multiple batch sizes should be separated by a ','.
+  -i, --ir IR                       location for the MLIR to be compiled, if compilation is done independently.
+  -p, --tensor-parallel {1,8}       tensor parallel size. Used for independent compilation. Defaults to 1.
+        """
+        _ = file.write(help_text + "\n")
+
+
+class BasePipeline:
+    def __init__(
+        self,
+        compile: bool,
+        shard: bool,
+        export: bool,
+        tp: int,
+        batch_sizes: list[int],
+        ir_path: Path | None,
+        artifacts_dir: Path,
+        weight_loc: Path,
+        export_dir: Path,
+    ):
+        self.compile: bool = compile
+        self.shard: bool = shard
+        self.export: bool = export
+        self.artifacts_dir: Path = artifacts_dir
+        self.weight_loc: Path = weight_loc
+        self.export_dir: Path = export_dir
+        self.mlir_path: Path | None = None
+        self.model_config_path: Path | None = None
+        self.export_batch_sizes: list[int] = batch_sizes
+        self.hip_target: str = "gfx942"
+        self.hal_target_backend: str = "rocm"
+        self.ir_path: Path | None = ir_path
+        self.tp: int = tp
+
+    def exec_pipeline(self):
+        self.validate()
+        if self.export:
+            self.export_tp1()
+
+        if self.shard:
+            self.export_tp8()
+
+        if self.compile:
+            self.compile_model()
+
+    def validate(self):
+        if not self.shard and not self.export and not self.ir_path and self.compile:
+            logger.error(
+                "To run compilation, either TP1 export (-e) or TP8 export (-s) must be specified, or path to IR must be passed in"
+            )
+            exit(1)
+
+        if not os.path.isfile(self.weight_loc):
+            logger.error(f"File {str(self.weight_loc)} does not exist")
+            raise ValueError(
+                f"Invalid path to weight file {str(self.weight_loc)}. File does not exist."
+            )
+
+        if not os.path.isdir(self.export_dir):
+            logger.warning(
+                f"Export directory {str(self.export_dir)} does not exist, creating..."
+            )
+            try:
+                Path.mkdir(self.export_dir, parents=True)
+            except:
+                logger.error("Failed to create export directory")
+                raise
+
+        if self.shard and not os.path.isdir(self.artifacts_dir):
+            logger.warning(
+                f"Artifacts directory {str(self.artifacts_dir)} does not exist, creating..."
+            )
+            try:
+                Path.mkdir(self.artifacts_dir, parents=True)
```

Comment:
same comments about `pathlib` usage
```suggestion
        if self.shard and not self.artifacts_dir.is_dir():
            logger.warning(
                f"Artifacts directory {str(self.artifacts_dir)} does not exist, creating..."
            )
            try:
                self.artifacts_dir.mkdir(parents=True)
```